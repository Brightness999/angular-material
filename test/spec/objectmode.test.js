/**
 * @fileoverview Test object mode of logging.
 */

const writeStream = require('flush-write-stream');

const Logality = require('../..');
const { stubLogality } = require('../lib/tester.lib');

describe('objectMode Logging', () => {
  stubLogality();

  test('Will return the Logging object', done => {
    const logality = new Logality({
      appName: 'testLogality',
      objectMode: true,
      wstream: writeStream.obj(chunk => {
        expect(typeof chunk).toEqual('object');
        expect(chunk).toMatchSnapshot();
        done();
      }),
    });

    const log = logality.get();

    log('info', 'hello world');
  });
});
