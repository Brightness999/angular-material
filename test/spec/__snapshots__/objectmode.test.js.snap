// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objectMode Logging Will return the Logging object 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
    },
    "source": Object {
      "file_name": "/test/spec/objectmode.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;
